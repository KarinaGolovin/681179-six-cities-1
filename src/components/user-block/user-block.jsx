import React from 'react';
import PropTypes from 'prop-types';

export const UserBlock = ({userAvatar, userName, userStatus}) => {
  return (
    <div className="property__host-user user">
      <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
        <img className="property__avatar user__avatar" src={`/${userAvatar}`} width="74" height="74" alt="Host avatar"/>
      </div>
      <span className="property__user-name">
        {userName}
      </span>
      {(userStatus) ? <span className="property__user-status">Pro</span> : null}
    </div>
  );
};

UserBlock.propTypes = {
  userAvatar: PropTypes.string,
  userName: PropTypes.string,
  userStatus: PropTypes.bool,
}

