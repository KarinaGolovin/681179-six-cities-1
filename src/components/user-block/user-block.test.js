import React from 'react';
import renderer from 'react-test-renderer';
import {UserBlock} from './user-block.jsx';

const mock = {
  name: `Test`,
  avatar: ``,
  status: false,
};

it(`UserBlock correctly renders`, () => {
  const component = renderer.create(
      <UserBlock
        userName={mock.name}
        userAvatar={mock.avatar}
        userStatus={mock.status}
      />
  ).toJSON();

  expect(component).toMatchSnapshot();
});
