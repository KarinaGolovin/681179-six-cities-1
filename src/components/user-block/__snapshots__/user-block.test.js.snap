// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock correctly renders 1`] = `
<div
  className="property__host-user user"
>
  <div
    className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
  >
    <img
      alt="Host avatar"
      className="property__avatar user__avatar"
      height="74"
      src="/"
      width="74"
    />
  </div>
  <span
    className="property__user-name"
  >
    Test
  </span>
</div>
`;
